### PR TITLE
ELF-32 and ELF-64 Comparison Reference

### PR DESCRIPTION
**Pull request by ejohnso49**
A reference post on the difference between 32-bit and 64-bit ELF

### CODE REVIEW

File: _drafts/elf-format-differences.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## ELF Background
+
+If you’re running a program on a UNIX-ish system, you’re probably using an ELF file. If you’re building firmware for your MCU, it’s probably an ELF file targeted for your MCU architecture. MacOS and Windows of course have their own slightly different formats, mach-o and Portable Executable. This format contains information on how to load your program into memory or to link with other object files to form an executable object. Static libraries are ELFs. The result of compiling a source file produces an ELF. Coredumps can be collected as ELFs. You might even be an ELF 😱!
```
Comment (noahp) [inline _drafts/elf-format-differences.md:22]:
```suggestion
If you’re running a program on a UNIX-ish system, you’re probably using an ELF file. If you’re building firmware for your MCU, it’s probably an ELF file targeted for your MCU architecture. MacOS and Windows of course have their own slightly different formats, Mach-O and Portable Executable. This format contains information on how to load your program into memory or to link with other object files to form an executable object. Static libraries are ELFs. The result of compiling a source file produces an ELF. Coredumps can be collected as ELFs. You might even be an ELF 😱!
```

File: _drafts/elf-format-differences.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+| Elf32_Word | 4 | 4 |
+| Elf32_Sword | 4 | 4 |
+| unsigned char | 1 | 1 |
+| - | - | - |
+| - | - | - |
+
+| Type | Size | Alignment |
+| --- | --- | --- |
+| Elf64_Addr | 8 | 8 |
+| Elf64_Off | 8 | 8 |
```
Comment (noahp) [inline _drafts/elf-format-differences.md:58]: Maybe combine in a single table, to better highlight the differences? Or add emphasis for the ones that are different?

**Review by noahp — APPROVED** [review]
`npx prettier --prose-wrap always --write` 🫶

Awesome post!

**Comment by tyhoff** [comment]
Forced a rebuild here to see if my Cloudflare preview build works.

**Pull request merged**